### PR TITLE
feat: using view transitions API on appearance toggle

### LINKED ISSUE #1270: [Feature]: 实现丝滑的动态切换主题 (using view transitions API on appearance toggle)
### 这个功能解决了什么问题？

大佬，如果想给站点实现如下效果，有什么方法吗？

我试了 globalUIComponents + globalStyles + 自定义监听点击事件，然后执行动画方法，并没有达到理想的效果！有什么方法能够拦截切换主题的事件，然后执行自定义事件吗？

https://github.com/user-attachments/assets/77c1d069-589b-4220-94f6-8c7c9e9b6829

### 你期望的 API 是什么样子的？

https://github.com/user-attachments/assets/77c1d069-589b-4220-94f6-8c7c9e9b6829

### PR DESCRIPTION
**Pull request by ayangweb**
## Summary

support `enableAppearanceAnimation` to use view transitions API on appearance toggle

## Related Issue

closes #1270

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by chenjiahan** [comment]
@SoonIter cc

**Comment by SoonIter** [comment]
this feature is conflicted with https://github.com/web-infra-dev/rspress/pull/101

two view-transition name is conflicted, 'flip' and 'root'

https://github.com/ayangweb/rspress/blob/5cfab69d674485933db69b09667576f416e96a58/packages/theme-default/src/layout/DocLayout/index.module.scss

when switching the dark mode, `.rspress-doc` with flip 's behavior is strange

Uploading QQ20240726-195058-HD.mp4…


Currently, I tried to solve the conflict but it does not work...

also need some help

**Comment by ayangweb** [comment]
@SoonIter 我想排查一下原因，可惜在我执行 `pnpm dev` 的时候，它经常报错！

image

**Comment by Timeless0911** [comment]
> @SoonIter 我想排查一下原因，可惜在我执行 `pnpm dev` 的时候，它经常报错！
> 
> image

你试着把 shared 包 和 这个包开两个终端 分别运行 dev

**Comment by SoonIter** [comment]
I solved it by some ugly codes

https://github.com/ayangweb/rspress/blob/cdc406d8532d13df53a7d851b4d1fec743061b4b/packages/theme-default/src/components/SwitchAppearance/index.tsx#L20

**Review by Timeless0911 — APPROVED** [review]

File: packages/document/rspress.config.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export default defineConfig({
   },
   themeConfig: {
     enableContentAnimation: true,
+    enableAppearanceAnimation: true,
```
Comment (Timeless0911) [inline packages/document/rspress.config.ts:49]:
```suggestion
    enableAppearanceAnimation: false,
```

**Comment by Timeless0911** [comment]
> [CLA assistant check](https://cla-assistant.io/web-infra-dev/rspress?pullRequest=1272) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/web-infra-dev/rspress?pullRequest=1272) before we can accept your contribution.**2** out of **3** committers have signed the CLA.✅ SoonIter✅ Timeless0911❌ ayangwebYou have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/web-infra-dev/rspress?pullRequest=1272) it.

@ayangweb Hi, can you sign the CLA to let CI pass~

**Comment by ayangweb** [comment]
> > [CLA assistant check](https://cla-assistant.io/web-infra-dev/rspress?pullRequest=1272) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/web-infra-dev/rspress?pullRequest=1272) before we can accept your contribution.**2** out of **3** committers have signed the CLA.✅ SoonIter✅ Timeless0911❌ ayangwebYou have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/web-infra-dev/rspress?pullRequest=1272) it.
> 
> @ayangweb Hi, can you sign the CLA to let CI pass~

Okay, I've signed it.

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**